lib/galaxy/webapps/galaxy/api/users.py

```diff
@@ -393,6 +394,60 @@ def set_information(self, trans, id, payload={}, **kwd):
         trans.log_event('User information added')
         return {'message': 'User information has been saved.'}
 
+    def send_verification_email( self, trans, email, username ):
+        """
+        Send the verification email containing the activation link to the user's email.
+        """
+        if username is None:
+            username = trans.user.username
+        activation_link = self.prepare_activation_link( trans, escape( email ) )
+
+        host = trans.request.host.split( ':' )[ 0 ]
+        if host in [ 'localhost', '127.0.0.1', '0.0.0.0' ]:
+            host = socket.getfqdn()
+        body = ("Hello %s,\n\n"
```

Comment:
The 'heavy' lifting here should be handled by e.g. managers. Web & API controllers should be lightweight, handling the formatting of inputs and outputs, with the bulk of the operations being performed by some centralized/shared methodology. 

This is a general comment and should not prevent merging the bug fix, however.